pkg/utils/download.go

```diff
@@ -12,34 +12,60 @@
 package utils
 
 import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
 	"net/url"
 	"os"
 	"path"
 	"strings"
 	"time"
+
+	"github.com/google/go-github/github"
+)
+
+type (
+	// GitCredentials : credentials to access GitHub or GitHubEnterprise
+	GitCredentials struct {
+		Username string `json:"username"`
+		Password string `json:"password"`
+	}
 )
 
 // DownloadFromURLThenExtract downloads files from a URL
 // to a destination, extracting them if necessary
-func DownloadFromURLThenExtract(URL string, destination string) error {
-	if _, err := url.ParseRequestURI(URL); err != nil {
+func DownloadFromURLThenExtract(URL, destination string, gitCredentials GitCredentials) error {
+	_url, err := url.ParseRequestURI(URL)
```

Comment:
I'm assuming `_url` was picked as a name to avoid a clash with `url`.

Reply:
yep, unfortunately!